packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/mira/MiraController.java

```diff
@@ -1,113 +1,140 @@
+import software.uncharted.terarium.hmiserver.models.dataservice.Artifact;
 import software.uncharted.terarium.hmiserver.models.dataservice.model.Model;
 import software.uncharted.terarium.hmiserver.models.task.TaskRequest;
 import software.uncharted.terarium.hmiserver.models.task.TaskResponse;
-import software.uncharted.terarium.hmiserver.models.task.TaskStatus;
 import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.data.ArtifactService;
 import software.uncharted.terarium.hmiserver.service.data.ModelService;
+import software.uncharted.terarium.hmiserver.service.tasks.MdlToStockflowResponseHandler;
+import software.uncharted.terarium.hmiserver.service.tasks.SbmlToPetrinetResponseHandler;
+import software.uncharted.terarium.hmiserver.service.tasks.StellaToStockflowResponseHandler;
 import software.uncharted.terarium.hmiserver.service.tasks.TaskService;
-import software.uncharted.terarium.hmiserver.service.tasks.TaskService.TaskType;
+import software.uncharted.terarium.hmiserver.service.tasks.TaskService.TaskMode;
 
 @RequestMapping("/mira")
 @RestController
 @Slf4j
 @RequiredArgsConstructor
 public class MiraController {
 
+	final private ArtifactService artifactService;
 	final private ObjectMapper objectMapper;
 	final private TaskService taskService;
 	final private ModelService modelService;
 
-	static final public String STELLA_TO_STOCKFLOW = "mira_task:stella_to_stockflow";
-	static final public String MDL_TO_STOCKFLOW = "mira_task:mdl_to_stockflow";
-	static final public String SBML_TO_PETRINET = "mira_task:sbml_to_petrinet";
-	static final private long REQUEST_TIMEOUT_SECONDS = 30;
-
 	@Data
 	static public class ModelConversionRequest {
-		public String modelName;
-		public String modelContent;
+		public UUID artifactId;
 	};
 
 	@Data
 	static public class ModelConversionResponse {
 		public Model response;
 	};
 
-	private boolean endsWith(String filename, List<String> suffixes) {
-		for (String suffix : suffixes) {
+	private boolean endsWith(final String filename, final List<String> suffixes) {
+		for (final String suffix : suffixes) {
 			if (filename.endsWith(suffix)) {
 				return true;
 			}
 		}
 		return false;
 	}
 
-	@PostMapping("/convert_and_create_model")
+	final private StellaToStockflowResponseHandler stellaToStockflowResponseHandler;
+	final private MdlToStockflowResponseHandler mdlToStockflowResponseHandler;
+	final private SbmlToPetrinetResponseHandler sbmlToPetrinetResponseHandler;
+
+	@PostConstruct
+	void init() {
+		taskService.addResponseHandler(stellaToStockflowResponseHandler);
+		taskService.addResponseHandler(mdlToStockflowResponseHandler);
+		taskService.addResponseHandler(sbmlToPetrinetResponseHandler);
+	}
+
+	@PostMapping("/convert-and-create-model")
 	@Secured(Roles.USER)
-	@Operation(summary = "Dispatch a mira conversion task")
+	@Operation(summary = "Dispatch a Mira conversion task")
```

Comment:
If you want to be pedantic... It's an acronym
```suggestion
	@Operation(summary = "Dispatch a MIRA conversion task")
```